asciidoc/modules/con_mg_migrating-clusters.adoc

```diff
@@ -0,0 +1,40 @@
+[id="migrating_clusters_{context}"]
+
+= Migrating clusters
+
+It is important to decide the migration strategy for your codebase. This is because you cannot add {VertX} {v3x} nodes and {VertX} {v4} nodes together in a single cluster for the following reasons:
+
+* Cluster manager upgrades - Major version upgrades in cluster managers prevent breaking (non-backward compatibility) changes.
+* Subscription data changes - {VertX} has changed the format of the EventBus subscription data stored in cluster managers.
+* Transport protocol changes - {VertX} has changed some fields in the message transport protocol in the cluster.
+
+If you have a {VertX} cluster for a single application or for some closely related microservices, you can migrate the entire codebase to the new cluster at one time.
+
+However, if you cannot migrate the codebase at one time, use the recommendations in this section to migrate a {VertX} {v3x} codebase to {VertX} {v4}.
+
+== Splitting the cluster
+
+If you have a cluster where different teams have deployed verticles for their applications, you can consider splitting the {VertX} {v3x} cluster into smaller ones. Note that after splitting the cluster, the separated components will not be able to communicate using the clustering features. You can split the cluster using the following:
+
+* EventBus request and reply - HTTP/RESTful web services, gRPC
+* EventBus send and publish - Messaging systems such as, {AMQ}, Postgres `LISTEN` and `NOTIFY`, Redis Pub and Sub
+* Shared Data - Redis, Infinispan, {Hazelcast}
```

Comment:
Shouldn't the comma and following space be part of the `Hazelcast` variable? Otherwise if the variable is not set there will be a trailing comma

Reply:
just remove the implementation names and avoid using variables for that.

Reply:
meaning that having a variable just to avoid hazelcast or redis looks awkward